inmem_transport.go

```diff
@@ -139,20 +139,27 @@ func (i *InmemTransport) makeRPC(target ServerAddress, args interface{}, r io.Re
 	}
 
 	// Send the RPC over
-	respCh := make(chan RPCResponse)
-	peer.consumerCh <- RPC{
+	respCh := make(chan RPCResponse, 1)
+	rpc := RPC{
 		Command:  args,
 		Reader:   r,
 		RespChan: respCh,
 	}
+	timer := time.After(timeout)
```

Comment:
Better to use time.NewTimer and defer timer.Stop, so that we don't leave excess timer objects when the timeout isn't triggered?